### PR TITLE
[PyTorch FE] Fix aten.index_add conversion for torch.export by handling missing alpha argument

### LINKED ISSUE #35628: [PyTorch FE] aten.index_add fails to convert via torch.export path because translator requires alpha input that torch.export elides
## OpenVINO Version

2026.1.0-21367-63e31528c62-releases/2026/1 (also reproduces against `master` HEAD as of 2026-04-30)

## Operating System

Linux x86_64 (Ubuntu, Python 3.12.3)

## Device used for inference

CPU (failure is at conversion time, before any device is chosen)

## Framework

PyTorch — `torch.export.export` → `openvino.convert_model`

## Model used

Minimal reproducer (attached as `reproducer.py`). Originally hit while exporting Mixtral 8x7B with a sparse top-2 expert dispatch path — the natural way to write the per-expert accumulator is `final.index_add_(0, tok_idx, contributions)`. The export-side workaround is to use `final.scatter_add(0, idx_b, contributions)` with a manually broadcast index tensor — see "Workaround" below.

## Issue description

`torch.export.export` produces an FX graph node with the call:

```python
torch.ops.aten.index_add.default(zeros, 0, tok_idx, contributions)
```

i.e. **four** positional inputs: `(self, dim, index, source)`. The PyTorch schema for `aten::index_add` is:

```
aten::index_add(Tensor self, int dim, Tensor index, Tensor source, *, Scalar alpha=1) -> Tensor
```

`alpha` is a keyword-only argument with default `1`. PyTorch's `torch.export` elides defaults from the FX graph (only positionally-supplied arguments appear), so the `default` overload arrives at the OpenVINO frontend with 4 inputs.

The OV translator at `src/frontends/pytorch/src/op/index_add.cpp:28` calls:

```cpp
num_inputs_check(context, 5, 6);
```

requiring at least 5 inputs — the 4 above plus `alpha`. With only 4 inputs supplied by `torch.export`, conversion aborts:

```
OpConversionFailure: Model wasn't fully converted. Failed operations detailed log:
-- aten.index_add.default with a message:
Exception happened during conversion of operation index_add/aten.index_add.default with schema NONE
Check 'num_inputs >= min_inputs' failed at src/frontends/pytorch/src/utils.cpp:48:
Got less inputs 4 than expected 5
```

The translator already calls `context.get_input(4)` unconditionally for `alpha` (line 33) and would multiply `src` by it — but with `torch.export` that input is simply not present in the graph.

The `jit.trace` path is unaffected because `jit.trace` materializes the default `alpha=1` into the captured graph as an explicit fifth argument. So the existing tests for `aten::index_add` (which run via the trace path) pass.

The same failure is registered for the in-place variant `aten.index_add_.default` via `op_table.cpp:983`, since both default and in-place variants route to the same translator that requires 5 inputs.

`aten.scatter_add.default` works correctly under `torch.export` and is what users currently fall back to. However, `scatter_add` requires the index tensor to have the same rank as the source (so users have to manually `index.unsqueeze(-1).expand_as(source)`), which is a non-trivial rewrite of code that would otherwise use the simpler `index_add` pattern.

## Step-by-step reproduction

```python
import openvino as ov
import torch
import torch.nn as nn


class IndexAddPath(nn.Module):
    def forward(self, hidden_states, tok_idx, contributions):
        bs, sq, hd = hidden_states.shape
        final = torch.zeros((bs * sq, hd), dtype=hidden_states.dtype, device=hidden_states.device)
        final = final.index_add(0, tok_idx, contributions)
        return final.view(bs, sq, hd)


m = IndexAddPath().eval()
ex_h = torch.randn(1, 4, 8)
ex_i = torch.tensor([0, 1, 2, 1], dtype=torch.long)
ex_c = torch.randn(4, 8)
S = torch.export.Dim("seq", min=1, max=64)
N = torch.export.Dim("nactive", min=1, max=128)

ep = torch.export.export(
    m,
    (ex_h, ex_i, ex_c),
    dynamic_shapes={
        "hidden_states": {1: S},
        "tok_idx": {0: N},
        "contributions": {0: N},
    },
)

# FX graph contains:
#   torch.ops.aten.index_add.default(zeros, 0, tok_idx, contributions)
# i.e. 4 positional inputs (alpha=1 default elided)

ov.convert_model(ep)  # raises OpConversionFailure
```

A self-contained side-by-side reproducer (also runs the working `scatter_add` path) is available as a gist: https://gist.github.com/t8/c9700b1a9b4e526cadc5be7e9dbe4510

## Relevant log output

```
torch=2.11.0+cu130  ov=2026.1.0-21367-63e31528c62-releases/2026/1

=== (A) index_add ===
  torch.export: OK
  FX graph (relevant lines):
    zeros = torch.ops.aten.zeros.default([sym_size_int_1, 8], dtype = torch.float32, device = device(type='cpu'), pin_memory = False)
    index_add = torch.ops.aten.index_add.default(zeros, 0, tok_idx, contributions)
    view = torch.ops.aten.view.default(index_add, [1, sym_size_int_1, 8])
  ov.convert_model FAILED: OpConversionFailure
    Check 'false' failed at src/frontends/common_translators/src/unconverted_ops_report.cpp:141:
    FrontEnd API failed with OpConversionFailure:
    Model wasn't fully converted. Failed operations detailed log:
    -- aten.index_add.default with a message:
    Exception happened during conversion of operation index_add/aten.index_add.default with schema NONE
    Check 'num_inputs >= min_inputs' failed at src/frontends/pytorch/src/utils.cpp:48:
    FrontEnd API failed with OpConversionFailure:
    Got less inputs 4 than expected 5

=== (B) scatter_add (workaround) ===
  ov.convert_model: OK (29 ops)
  CPU compile: OK
```

## Suggested fix

Two-line change at `src/frontends/pytorch/src/op/index_add.cpp`:

```diff
 OutputVector translate_index_add(const NodeContext& context) {
-    num_inputs_check(context, 5, 6);
+    num_inputs_check(context, 4, 6);
     auto input = context.get_input(0);
     auto dim = context.get_input(1);
     auto index = context.mark_node(std::make_shared<v0::Convert>(context.get_input(2), element::i32));
     auto src = context.get_input(3);
-    auto alpha = context.get_input(4);
+    Output<Node> alpha;
+    if (context.get_input_size() < 5 || context.input_is_none(4)) {
+        alpha = context.mark_node(v0::Constant::create(element::f32, Shape{}, {1.0}));
+    } else {
+        alpha = context.get_input(4);
+    }
     auto converted_alpha = context.mark_node(std::make_shared<v1::ConvertLike>(alpha, src));
     auto alpha_src = context.mark_node(std::make_shared<v1::Multiply>(converted_alpha, src));
```

The `ConvertLike` already brings the alpha constant to `src`'s element type, so creating it as `f32` is fine. Happy to open a PR if the team agrees with the approach.

This same pattern (translator requires keyword-default input that `torch.export` elides) likely affects other translators registered for `*.default` overloads in `op_table.cpp`. A grep for `num_inputs_check(context, N` where `N` is greater than the count of mandatory positional args in the corresponding ATen schema would surface them.

## Related issues

None found via GitHub search:
- `index_add num_inputs`, `aten.index_add torch.export`, `index_add OpConversionFailure`, `Got less inputs`, `index_add default alpha` — all returned 0 hits in `openvinotoolkit/openvino` issues and PRs.

## Checklist

- [x] Reproduces on the latest released OpenVINO (2026.1.0).
- [x] Reproduces on `master` (translator code unchanged between 2026.1 release and HEAD).
- [x] Self-contained reproducer attached.
- [x] Suggested fix is single-file (`src/frontends/pytorch/src/op/index_add.cpp`).

### PR DESCRIPTION
**Pull request by andersendsa**
### Details:
When exporting PyTorch models containing `aten.index_add` using `torch.export.export`, the resulting FX graph elides the default keyword-only argument `alpha=1`. As a result, the node arrives at the OpenVINO PyTorch frontend with only 4 positional inputs instead of the expected 5. This causes the conversion to fail with an `OpConversionFailure`.
closes: #35628
### Changes
* **Relaxed input constraints:** Updated `num_inputs_check` in `src/frontends/pytorch/src/op/index_add.cpp` to require a minimum of 4 inputs instead of 5.
* **Handled missing alpha:** Added fallback logic to check if the `alpha` argument (at index 4) is missing or `None`. If so, it dynamically creates a default `f32` scalar constant of `1.0`.

### AI Assistance:
 - *AI assistance used: no

### CODE REVIEW

**Comment by andersendsa** [comment]
Hi @mmikolajcz , @PiotrKrzem  could you pls let me know if this pr needs any changes or if it is good to merge

**Review by rkazants — CHANGES_REQUESTED** [review]
Please add layer tests

**Comment by andersendsa** [comment]
> Please add layer tests

Hi @rkazants  I have added layer tests and this pr is ready for review

**Comment by andersendsa** [comment]
Hi @mlukasze could u you pls review this pr

**Comment by he-ub** [comment]
Hi @andersendsa, is there any updates on this PR please ?

**Comment by andersendsa** [comment]
> Hi @andersendsa, is there any updates on this PR please ?

Hi @he-ub  i belive it is good to merge pls review it

**Comment by he-ub** [comment]
Hi @PiotrKrzem @mmikolajcz @rkazants, do you think we can merge this ?

**Comment by he-ub** [comment]
> Hi @he-ub i belive it is good to merge pls review it

LGTM.
